crates/store/src/db/mod.rs

```diff
@@ -269,7 +270,7 @@ impl Db {
         let conn = pool.get().await.map_err(DatabaseError::MissingDbConnection)?;
 
         conn.interact(apply_migrations).await.map_err(|err| {
-            DatabaseError::InteractError(format!("Migration task failed: {err}"))
+            DatabaseError::InteractError(format!("Migration task failed: {}", err.as_report()))
```

Comment:
@Mirko-von-Leipzig Do we want to handle errors like `DatabaseError::InteractError(String)` like this or modify them to be `DatabaseError::InteractError(anyhow::Error)` and do
```rust
            DatabaseError::InteractError(anyhow!("Migration task failed: {err}"))
```

Reply:
cc @PhilippGackstatter

Reply:
> @Mirko-von-Leipzig Do we want to handle errors like DatabaseError::InteractError(String) like this or modify them to be DatabaseError::InteractError(anyhow::Error) and do

Yeah; though you'll find `anyhow::Error` isn't send. So you need to use `Arc<anyhow::Error>` which somewhat breaks the convenience of using `anyhow`.

```rust
DatabaseError::InteractError(anyhow!("Migration task failed: {err}"))
```

^^ this also doesn't work as you might expect because `{err}` just uses the display of the error and misses out on the error reporting. What one has to do is either `err.context("migration task failed")`, or use the `err.as_report()`. The former would be the `anyhow` way but now you also need to `Arc` it (I suspect).

Reply:
Less manual:
```rust
pub trait ErrorReport: std::error::Error {
    fn as_wrapped_report(&self, msg: &'static str) -> String {
        format!("{msg}: {}", self.as_report())
    }
}
...
DatabaseError::InteractError(err.as_wrapped_report("Migration task failed"))
...
```

Or it returns `Error` and you need to `.to_string()` manually.

WDYT?

Reply:
So I guess lets go with your `as_wrapped_report` (whichever impl you care to use). And we can punt the R&D a bit.